core/src/main/java/org/apache/accumulo/core/clientImpl/AccumuloClientImpl.java

```diff
@@ -283,12 +287,21 @@ public void close() {
     }
   }
 
-  public static class AccumuloClientBuilderImpl
-      implements InstanceArgs, PropertyOptions, ClientInfoOptions, AuthenticationArgs,
-      ConnectionOptions, SslOptions, SaslOptions, AccumuloClientFactory, FromOptions {
+  public static class ClientBuilderImpl<T>
+      implements InstanceArgs<T>, PropertyOptions<T>, AuthenticationArgs<T>, ConnectionOptions<T>,
+      SslOptions<T>, SaslOptions<T>, ClientFactory<T>, FromOptions<T> {
 
     private Properties properties = new Properties();
     private AuthenticationToken token = null;
+    private BuildFunction<T> builderFunction;
```

Comment:
The `BuildFunction` interface isn't needed. Can just use `Function<ClientBuilderImpl, T>` here.

Reply:
Done